core/app/models/spree/stock/content_item.rb

```diff
@@ -36,6 +36,17 @@ def amount
         price * quantity
       end
 
+      def hash
+        # XOR operation to get a unique combo hash
+        inventory_unit.hash ^ state.hash
+      end
+
+      def eql?(other)
+        other.class == self.class &&
+          other.inventory_unit == inventory_unit &&
+          (!other.state || other.state == state)
```

Comment:
Have a look at the answer above from @gmacdougall where he explains a `change in behaviour`. I had it initially as just `other.state == state`, but that changes the behaviour as explained by Gregor, so I copied the original.